### ISSUE TITLE
Using rmarkdown::render instead of wflow_build

### ISSUE THREAD

**Issue by jennysjaarda** [jdblischak/workflowr#260]
I am using `workflowr` inside a [`targets`](https://books.ropensci.org/targets/) (@wlandau) project and I am trying to take advantage of the convenient `tar_render` function from `targets`. As far as I can tell, `tar_render` is basically a wrapper around `rmarkdown::render` but with some added features like looking for outdated data inside the Rmd report to know if the report should be generated again. I would really like to use this function within `workflowr`, but am running into an issue. 

If I use `rmarkdown::render()` on my `workflowr` created Rmd documents, I get an error:

```
rmarkdown::render("analysis/about.Rmd", output_dir = "docs")

output file: about.knit.md

/usr/bin/pandoc +RTS -K512m -RTS about.knit.md --to html4 --from markdown+autolink_bare_uris+tex_math_single_backslash --output /data/sgg2/jenny/projects/proxyMR/docs/about.html --self-contained --variable bs3=TRUE --standalone --section-divs --template /data/sgg2/jenny/projects/proxyMR/renv/library/R-4.1/x86_64-pc-linux-gnu/rmarkdown/rmd/h/default.html --no-highlight --variable highlightjs=1 --include-before-body /tmp/RtmpoQq0Ik/rmarkdown-str1267e14a386e8.html --variable navbar=1 --variable theme=bootstrap --include-in-header /tmp/RtmpoQq0Ik/rmarkdown-str1267e6c045df7.html --mathjax --variable 'mathjax-url:https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML' --include-in-header /tmp/RtmpoQq0Ik/header1267e72eb8aa0.html --include-after-body /tmp/RtmpoQq0Ik/footer1267e64eaefcd.html 
Fetching https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png...
pandoc: Failed to retrieve https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png
user error (https not supported)
Error: pandoc document conversion failed with error 61

```

But if I change the header from `output: workflowr::wflow_html` to `output: html_document` *and* remove the following from the `_site.yml` file. I can generate my html document. 

```
  right:
  - icon: fa-github
    text: Source code
    href: https://github.com/jennysjaarda/proxyMR
output:
  workflowr::wflow_html:
    toc: yes
    toc_float: yes
    theme: flatly
    highlight: textmate
    number_sections: true
```

Unfortunately this isn't a great solution because I've lost all the `workflowr` benefits. Do you have any suggestions for how to get `rmarkdown::render` to work on a `workflowr` document? 

I think the best solution would be to stick with `rmarkdown::render_site`, but this doesn't work with `targets` unfortunately unless @wlandau is willing to consider adding this functionality :)

See thread here: https://github.com/ropensci/targets/discussions/648

Thanks in advance for your suggestions!

**Comment by jennysjaarda** [comment]
And just to followup, I tried to use the `render_site` function on my Rmd files: `rmarkdown::render_site("analysis/about.Rmd")`, but realize this is not really optimal either because then the site isn't built in a "fresh R session" and we get this warning on the rendered html. Open to any suggestions either of you may have! 
![image](https://user-images.githubusercontent.com/36084054/138087101-ed0b0ce5-d17f-41b0-a4a3-0c929eaa0c47.png)

**Comment by wlandau** [comment]
Maybe `callr::r(function() rmarkdown::render_site("analysis/about.Rmd"))`?

**Comment by jennysjaarda** [comment]
Thanks! If I do that, then all checks pass within `workflowr`! But I still would like to find a solution to somehow integrate this with `tar_render` somehow, which I think I need to get it to work using `rmarkdown::render` not `rmarkdown::render_site`... any suggestions on that?

**Comment by wlandau** [comment]
There does not exist a `tar_render()` for `render_site()`. Once we start thinking about all these specific cases - `workflowr`, R Markdown sites, `bookdown`, `blogdown` - the scope explodes and there is a lot to maintain. As a workaround you could define collections of targets to register dependencies you know show up in your site.

```{r}
# _targets.R
list(
  # ...,
  tar_target(files, c("site.Rmd", "style.css"), format = "file"),
  tar_target(site, {
    # declare dependencies
    files
    loaded_target1
    loaded_target2
    # render the site
    rmarkdown::render_site("site.Rmd")
    # return the tracked output files
    "site.html"
  }, format = "file")
)

```

**Comment by jdblischak** [comment]
Thanks @jennysjaarda for the thorough bug report and thanks @wlandau for your troubleshooting assistance.

Trying to follow everything in this thread and https://github.com/ropensci/targets/discussions/648, I'm having trouble understanding what the root issue is. Other workflowr users have integrated targets into their projects, so I am pretty sure this is possible, e.g. see #238 and https://github.com/ropensci/tarchetypes/issues/23

It seems like there are some network issues with pandoc. The error in the original post in this thread was because pandoc couldn't download the icon that workflowr puts in the browser tab. And in https://github.com/ropensci/targets/discussions/648, there were issues downloading the fa-github icon. I think we need to first rule out pandoc problems before trying to troubleshoot workflowr+targets.

@jennysjaarda Could you please run `rmarkdown::render()` on the Rmd file below? It will test whether your pandoc can download the same web resources as those used in a standard workflowr project.

````
---
title: "Test pandoc network issues"
output: html_document
---

This is the workflowr icon 
![workflowr](https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png)

```{r fontawesome, include=FALSE}
# https://github.com/rstudio/rmarkdown/issues/813#issuecomment-248792317
htmltools::tagList(rmarkdown::html_dependency_font_awesome())
```

This is the fontawesome GitHub icon 
<i class="fa fa-github"></i>

````
